packages/chaire-lib-backend/src/config/server.config.ts

```diff
@@ -354,6 +360,22 @@ export const setProjectConfiguration = (newConfig: Partial<ProjectConfiguration<
         });
     }
 
+    // Make sure the maximum number of parallel calculators is a positive value, but does not exceed the CPU count
+    if (newConfig.maxParallelCalculators !== undefined) {
+        const availableCPUs = os.cpus().length;
+        if (newConfig.maxParallelCalculators <= 0) {
+            console.warn(
+                `maxParallelCalculators (${newConfig.maxParallelCalculators}) must be a positive number. Using the number of CPUs instead: ${availableCPUs}`
+            );
+            newConfig.maxParallelCalculators = availableCPUs;
+        } else if (newConfig.maxParallelCalculators > availableCPUs) {
+            console.warn(
+                `maxParallelCalculators (${newConfig.maxParallelCalculators}) cannot exceed the number of CPUs. Using the number of CPUs instead: ${availableCPUs}`
+            );
+            newConfig.maxParallelCalculators = availableCPUs;
```

Comment:
We should let users set a value higher than the currently available CPU. 
Could be test purposes or other. 
We can keep the warning, but not override the setting